homeassistant/components/tts/media_source.py

```diff
@@ -148,26 +157,28 @@ async def async_browse_media(
         )
 
     @callback
-    def _provider_item(
-        self, provider_domain: str, params: str | None = None
-    ) -> BrowseMediaSource:
+    def _engine_item(self, engine: str, params: str | None = None) -> BrowseMediaSource:
         """Return provider item."""
-        manager: SpeechManager = self.hass.data[DOMAIN]
-        if (provider := manager.providers.get(provider_domain)) is None:
+        if (engine_instance := get_engine_instance(self.hass, engine)) is None:
             raise BrowseError("Unknown provider")
 
+        if engine.startswith(f"{DOMAIN}."):
+            engine_domain = split_entity_id(engine)[1]
```

Comment:
We would want to make the name configurable in a future PR, so let's avoid already relying on it being the domain.

```suggestion
            engine_domain = engine.platform.config_entry.domain
```